apps/wallet/src/features/Widget/BuiltInNetwork/TokenTab.tsx

```diff
@@ -1,18 +1,25 @@
 import type { FC } from 'react';
 import { StyleSheet } from 'react-native';
-import type { Networks } from '@walless/core';
+import type { Networks, Token } from '@walless/core';
+import type { TokenDocument } from '@walless/store';
 import { useTokens } from 'utils/hooks';
 
 import TokenList from './TokenList';
 
 interface Props {
 	network: Networks;
+	tokens?: TokenDocument<Token>[];
 }
 
-export const TokenTab: FC<Props> = ({ network }) => {
-	const { tokens } = useTokens(network);
+export const TokenTab: FC<Props> = ({ network, tokens }) => {
+	const { tokens: networkTokens } = useTokens(network);
 
-	return <TokenList items={tokens} style={styles.tokenListContainer} />;
+	return (
+		<TokenList
+			items={tokens ? tokens : networkTokens}
+			style={styles.tokenListContainer}
+		/>
+	);
```

Comment:
you are trying to make this `TokenTab` act like a Component and binding to internal state from hook `useTokens`, I prefer cutting it to stateless component without `useTokens` inside and totally pass tokens as a prop